test/CoreTypechecker.hs

```diff
@@ -263,6 +294,9 @@ kAppICompTy =
 -- ((Nat -> Nat -> Nat) -> (Nat -> Nat -> Nat) -> (Nat -> Nat -> Nat)) ->
 -- ((Nat -> Nat -> Nat) -> Nat -> Nat -> Nat)
 -- (Nat -> Nat -> Nat)
+
+-- S (K11) = (K 1) (1 1) doesn't work because one cannot properly annotate the second input of K (1 1).
```

Comment:
`S` in general might require first-class usages (luckily, we're working on that!)

However, it should be possible to write *specialised* versions of `S` for certain usages (but any type) right now.

Reply:
Woohoo first-class usages!

I'm not sure how to annotate 

1 applied to 1

but I have to annotate that because I have to annotate the inputs to `K` and `(1 1)` is the second input to `K`. Any idea?

Reply:
Ah I see - in that case, the first `1` is `((a -> a) -> (a -> a)) -> (a -> a) -> (a -> a)` and the second is `(a -> a) -> a -> a` (like usual) - the `a` in the first is just `a -> a`.

Reply:
Yes, I could try Church-encoded numbers. I can also test `S K I 1`.

Reply:
Ah sorry, I thought you were trying Church-encoded numbers - `1 1` with naturals doesn't make too much sense - `S K I` would be good to test.